src/main/java/edu/umich/ctools/sectionsUtilityTool/FriendServlet.java

```diff
@@ -17,235 +17,353 @@
-			throws IOException {
-		String pathInfo = request.getPathInfo();
-		PrintWriter out = response.getWriter();
-		JsonObject json;
+        M_log.debug("doPOST: Called");
+        boolean socialLoginEnabled = Boolean.parseBoolean(appExtProperties.getProperty(Utils.ENABLE_SOCIAL_LOGIN));
+        if (socialLoginEnabled) {
+            socialLoginRestApiCall(request, response);
+        } else {
+            friendRestApiCall(request, response);
+        }
+    }
+
+    private void socialLoginRestApiCall(HttpServletRequest request,
+                                        HttpServletResponse response) {
+        M_log.debug("socialLoginRestApiCall(): called");
+        long startTime = System.currentTimeMillis();
+        PrintWriter out = null;
+        JsonObject json;
+        String key = appExtSecureProperties.getProperty(Utils.SOCIAL_LOGIN_API_KEY);
+        String secret = appExtSecureProperties.getProperty(Utils.SOCIAL_LOGIN_API_SECRET);
+        String url = appExtSecureProperties.getProperty(Utils.SOCIAL_LOGIN_API_URL);
+        String spURL = appExtProperties.getProperty(Utils.SOCIAL_LOGIN_API_SERVICE_PROVIDED_URL);
+        String fullSocialLoginApiUrl = url + "/guest/invite";
+
+        try {
+            response.setContentType("application/json");
+            request.setCharacterEncoding(Utils.CONSTANT_UTF_8);
+            StringEntity apiPostBodyStringEntity;
+            out = response.getWriter();
+            Utils.logApiCall(request);
+
+            String nonUmichEmail = URLEncoder.encode(request.getParameter(PARAMETER_ID), Utils.CONSTANT_UTF_8).replace("%40", "@");
+            String nonUmichEmailFirstName = URLEncoder.encode(request.getParameter(PARAMETER_FRIEND_FIRST_NAME), Utils.CONSTANT_UTF_8);
+            String nonUmichEmailLastName = URLEncoder.encode(request.getParameter(PARAMETER_FRIEND_LAST_NAME), Utils.CONSTANT_UTF_8);
+            String keySecretStr = String.format("%s:%s", key, secret);
+
+            String encoding = Base64.getEncoder().encodeToString((keySecretStr).getBytes());
+            CloseableHttpClient httpclient = HttpClients.createDefault();
+            HttpPost httpPost = new HttpPost(fullSocialLoginApiUrl);
+            httpPost.setHeader("Accept", Utils.APPLICATION_JSON);
+            httpPost.setHeader("Content-type", Utils.APPLICATION_JSON);
+            httpPost.setHeader(HttpHeaders.AUTHORIZATION, "Basic " + encoding);
+            apiPostBodyStringEntity = socialLoginApiRequestBody(spURL, nonUmichEmail, nonUmichEmailFirstName, nonUmichEmailLastName);
+            httpPost.setEntity(apiPostBodyStringEntity);
+            HttpResponse res = httpclient.execute(httpPost);
+            if (res == null) {
+                response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+                json = buildResponseObject("error", "Non-Umich Account NOT created for due to No response", fullSocialLoginApiUrl);
```

Comment:
Need to log the proposed parameters, e.g.  nonUmichEmail, nonUmichEmailFirstName, etc.

Reply:
I have a fix for this here https://github.com/tl-its-umich-edu/canvas_course_manager/pull/266/commits/484e2800e5f801745ae1f98df6c19d06cd8e90d8#r346471959